src/main/java/com/quantumsoft/qupathcloud/repository/Repository.java

```diff
@@ -106,10 +108,7 @@
     cloudDao.set(new CloudDaoImpl(oAuth20));
 
     hierarchyListener = new QuPathHierarchyListener();
-    qupath.addImageDataChangeListener((source, imageDataOld, imageDataNew) -> {
-      LOGGER.trace("ImageData change old: " + imageDataOld + ", new:" + imageDataNew);
-      hierarchyListener.setListenedImageData(imageDataNew);
```

Comment:
we are not setting this ImageListenedData in the hierarchy listener

Reply:
I moved this code to the following part:
```
    @Override
    public void changed(ObservableValue<? extends ImageData<BufferedImage>> source,
        ImageData<BufferedImage> imageDataOld, ImageData<BufferedImage> imageDataNew) {
      LOGGER.trace("ImageData change old: " + imageDataOld + ", new:" + imageDataNew);
      if (this.imageData != null) {
        this.imageData.getHierarchy().removePathObjectListener(this);
      }
      if (imageDataNew != null) {
        this.imageData = imageDataNew;
        this.imageData.getHierarchy().addPathObjectListener(this);
      }
    }
```

Reply:
LGTM